src/cljs/org/broadinstitute/firecloud_ui/page/workspace/submission_details.cljs

```diff
@@ -126,9 +126,43 @@
                    (:workflows props))}])))})
 
 
+(react/defc AbortButton
+  {:kill-submission
+   (fn [{:keys [props state this]}]
+     (swap! state assoc :stopping-submission? true)
+     (let [submission-id (get (:submission props) "submissionId" )
+           workspace-id (:workspace-id props)
+           path (paths/submission-kill workspace-id submission-id)
+           headers {"Content-Type" "application/json"}
+           canned-response {:responseText ""
+                            :status 200 :delay-ms (rand-int 2000)}
+           on-done (fn [{:keys [success? status-text]}]
+                     (swap! state assoc :stopping-submission? false)
+                     (if success?
+                       ((:on-kill props))
+                       (js/alert (str "Error in stopping the job : " status-text))))]
+       (utils/ajax-orch path
+         {:headers headers
+          :method "DELETE"
+          :canned-response canned-response
+          :on-done on-done})))
+   :render (fn [{:keys [props state this]}]
+             (if (:stopping-submission? @state)
+               [comps/Blocker {:banner "Stopping submission ..."}]
+                (when (= "Submitted" ((:submission props) "status"))
```

Comment:
I think this logic should be moved to where AbortButton is created, on line 181.  Don't even create an AbortButton instance unless the status is "Submitted".